scene/gui/text_edit.cpp

```diff
@@ -105,7 +129,11 @@ int TextEdit::Text::get_line_height(int p_line, int p_wrap_index) const {
 }
 
 void TextEdit::Text::set_width(float p_width) {
+	if (width == p_width) {
+		return;
+	}
 	width = p_width;
+	is_dirty = true;
```

Comment:
`set_width` and `set_tab_size` do not fully invalidate cache, it's calling `invalidate_all_lines()` instead (which resetting line breaks only).